src/DataFramesMeta.jl

```diff
@@ -392,11 +392,44 @@ function transform(g::GroupedDataFrame; kwargs...)
     idx1 = [1; 1 .+ idx2[1:end-1]]
     for (k, v) in kwargs
         first = v(g[1])
-        result[k] = Array{eltype(first)}(undef, size(result, 1))
-        result[idx1[1]:idx2[1], k] = first
-        for i in 2:length(g)
-            result[idx1[i]:idx2[i], k] = v(g[i])
+        if first isa AbstractVector 
+            if length(first) != size(g[1], 1)
+                throw("If a function returns a vector, the result " * 
+                      "must have the same length as the groups it " *
+                      "operates on")
+            end
+            t = Tables.allocatecolumn(eltype(first), size(result, 1))
+            t[idx1[1]:idx2[1]] = first
+            for i in 2:length(g)
+                out = v(g[i])
+                if length(out) != size(g[i], 1)
+                    throw("If a function returns a vector, the result " * 
+                      "must have the same length as the groups it " *
+                      "operates on")
+                end
+                S = eltype(out)
+                T = eltype(t)
+                if !(S <: T || promote_type(S, T) <: T)
+                    t = copyto!(Tables.allocatecolumn(promote_type(S, T), size(result, 1)), 
+                        t[1:idx2[i-1]])
```

Comment:
That's not what I meant (this will create a copy). Do `copy!(Tables.allocatecolumn(...), 1, t, 1, idx2[i-1])` (hopefully I got it right).

Also fix indentation.

Reply:
Ah i see now. Thanks. Fixed this.